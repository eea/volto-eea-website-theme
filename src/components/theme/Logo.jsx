import { useSelector } from 'react-redux';
import { defineMessages, useIntl } from 'react-intl';
import config from '@plone/volto/registry';

import { Logo } from '@eeacms/volto-eea-design-system/ui';
import LogoImage from '@eeacms/volto-eea-design-system/../theme/themes/eea/assets/images/Header/eea-logo.svg';

const messages = defineMessages({
  site: {
    id: 'Site',
    defaultMessage: 'Site',
  },
  eeasite: {
    id: 'European Environment Agency',
    defaultMessage: 'European Environment Agency',
  },
});

const EEALogo = () => {
  const lang = useSelector((state) => state.intl.locale);
  const intl = useIntl();
  const url = config.settings.isMultilingual ? `/${lang}` : '/';

  return (
    <Logo
      src={LogoImage}
      url={url}
      title={intl.formatMessage(messages.site)}
      alt={intl.formatMessage(messages.eeasite)}
    />
  );
};

export default EEALogo;
