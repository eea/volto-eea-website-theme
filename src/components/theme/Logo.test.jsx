import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-intl-redux';

import Logo from './Logo';

const mockStore = configureStore();
let history = createMemoryHistory();

describe('Logo', () => {
  it('renders a Logo component', () => {
    const store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
      router: {
        location: {
          pathname: '/',
        },
      },
    });

    const component = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <Logo />
        </Router>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
