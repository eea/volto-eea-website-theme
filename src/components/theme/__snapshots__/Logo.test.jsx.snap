// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders a Logo component 1`] = `
<a
  className="logo"
  href="/"
  onClick={[Function]}
  title="Site"
>
  <img
    alt="European Environment Agency"
    className="ui image eea-logo"
    src="eea-logo.svg"
    title="Site"
  />
</a>
`;
